### ISSUE TITLE
Active consent to Terms of Service when signing up with Google

### ISSUE THREAD

**Issue by juanlaube** [toggl/mobileapp#3684]
🙊 🙈 🙉 

When signing up with Google, we’re not actively consenting to terms of service like we do with a sign up with email. This means we're not sending `tos_accepted` and `country_id` in the `/signup?app_name=toggl_mobile` request and the API returns BadRequest (luckily, otherwise we would not be complying with GDPR 🤔).

We need to:
- Present terms of service and privacy policy modal before starting sign up with Google
- Validate we have a selected country, also before starting sign up with Google
- Send `tos_accepted` and `country_id` in the request to Toggl API

**Comment by amulware** [comment]
Do I understand correctly, you propose to do country/tos/pp stuff before doing any interaction with google?

Does this need any sort of designs?

**Comment by juanlaube** [comment]
Yeah, that's what I propose because accepting ToS after the user gives permission to Toggl in their google account doesn't seem right. Maybe product people can input on this 🤔.

Doesn't need any sort of design because the ToS and PP modal is the same we already have for sign up with email.

**Comment by amulware** [comment]
No further approval needed. Let's do this.
